test/dag_test.go

```diff
@@ -196,71 +199,48 @@ func TestDAGPipelineRun(t *testing.T) {
 }
 
-func (f runTimes) Less(i, j int) bool {
-	return f[i].t.Before(f[j].t)
-}
-
-func (f runTimes) Swap(i, j int) {
-	f[i], f[j] = f[j], f[i]
-}
+func verifyExpectedOrder(t *testing.T, c clientset.TaskRunInterface) {
+	t.Logf("Verifying order of execution")
 
-func getTaskStartTimes(t *testing.T, c clientset.TaskRunInterface) runTimes {
-	taskRuns, err := c.List(metav1.ListOptions{})
+	taskRunsResp, err := c.List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("Couldn't get TaskRuns (so that we could check when they executed): %v", err)
 	}
-	times := runTimes{}
-	for _, t := range taskRuns.Items {
-		times = append(times, runTime{
-			name: t.Name,
-			t:    t.Status.StartTime.Time,
-		})
-	}
-	return times
-}
-
-func vefifyExpectedOrder(t *testing.T, times runTimes) {
-	if len(times) != 5 {
-		t.Fatalf("Expected 5 Taskruns to have executed but only got start times for %d Taskruns", len(times))
+	taskRuns := taskRunsResp.Items
+	if len(taskRuns) != 5 {
+		t.Fatalf("Expected 5 TaskRuns to have executed but only got start times for %d TaskRuns", len(taskRuns))
 	}
 
-	sort.Sort(times)
+	sort.Slice(taskRuns, func(i, j int) bool {
+		it := taskRuns[i].Status.StartTime.Time
+		jt := taskRuns[j].Status.StartTime.Time
+		return it.Before(jt)
+	})
 
-	if !strings.HasPrefix(times[0].name, "dag-pipeline-run-pipeline-task-1") {
-		t.Errorf("Expected first task to execute first, but %q was first", times[0].name)
+	if !strings.HasPrefix(taskRuns[0].Name, "dag-pipeline-run-pipeline-task-1") {
+		t.Errorf("Expected first task to execute first, but %q was first", taskRuns[0].Name)
 	}
-	if !strings.HasPrefix(times[1].name, "dag-pipeline-run-pipeline-task-2") {
-		t.Errorf("Expected parallel tasks to run second & third, but %q was second", times[1].name)
+	if !strings.HasPrefix(taskRuns[1].Name, "dag-pipeline-run-pipeline-task-2") {
+		t.Errorf("Expected parallel tasks to run second & third, but %q was second", taskRuns[1].Name)
 	}
-	if !strings.HasPrefix(times[2].name, "dag-pipeline-run-pipeline-task-2") {
-		t.Errorf("Expected parallel tasks to run second & third, but %q was third", times[2].name)
+	if !strings.HasPrefix(taskRuns[2].Name, "dag-pipeline-run-pipeline-task-2") {
+		t.Errorf("Expected parallel tasks to run second & third, but %q was third", taskRuns[2].Name)
 	}
-	if !strings.HasPrefix(times[3].name, "dag-pipeline-run-pipeline-task-3") {
-		t.Errorf("Expected third task to execute third, but %q was third", times[3].name)
+	if !strings.HasPrefix(taskRuns[3].Name, "dag-pipeline-run-pipeline-task-3") {
+		t.Errorf("Expected third task to execute third, but %q was third", taskRuns[3].Name)
 	}
-	if !strings.HasPrefix(times[4].name, "dag-pipeline-run-pipeline-task-4") {
-		t.Errorf("Expected fourth task to execute fourth, but %q was fourth", times[4].name)
+	if !strings.HasPrefix(taskRuns[4].Name, "dag-pipeline-run-pipeline-task-4") {
+		t.Errorf("Expected fourth task to execute fourth, but %q was fourth", taskRuns[4].Name)
```

Comment:
Is `dag-pipeline-run-pipeline-task-2` supposed to be there twice on line 226 and 229?
Also, can we put this in a loop?

Reply:
oh, I see what this is doing. 🤦 nvm.

Reply:
Yeah it confused me too, and even after I "got" it I had a moment of panic when I read your comment that this test wasn't correct. I think we could use better names :)